scipy/signal/_signaltools.py

```diff
@@ -2178,22 +2178,26 @@ def lfilter(b, a, x, axis=-1, zi=None):
     if zi is not None:
        zi = np.asarray(zi)
 
+    if not (b.ndim == 1 and xp_size(b) > 0):
+        raise ValueError(f"Parameter b is not a non-empty 1d array, since {b.shape=}!")
+    if not (a.ndim == 1 and xp_size(a) > 0):
+        raise ValueError(f"Parameter a is not a non-empty 1d array, since {a.shape=}!")
+
     if len(a) == 1:
         # This path only supports types fdgFDGO to mirror _linear_filter below.
         # Any of b, a, x, or zi can set the dtype, but there is no default
         # casting of other types; instead a NotImplementedError is raised.
         b = np.asarray(b)
         a = np.asarray(a)
-        if b.ndim != 1 and a.ndim != 1:
-            raise ValueError('object of too small depth for desired array')
         x = _validate_x(x)
         inputs = [b, a, x]
         if zi is not None:
             # _linear_filter does not broadcast zi, but does do expansion of
             # singleton dims.
             zi = np.asarray(zi)
             if zi.ndim != x.ndim:
-                raise ValueError('object of too small depth for desired array')
+                raise ValueError("Dimensions of parameters x and zi must match, but " +
+                                 f"{x.ndim=}, {zi.ndim=}!")
```

Comment:
I think this should be something like:
```python
f"x.ndim={x.ndim}, zi.ndim={zi.ndim}!")
```

Reply:
Doesn't produce that the same output? E.g.:
```python
>>> import numpy as np
>>> x, zi = np.arange(5), np.eye(3)
>>> f"{x.ndim=}, {zi.ndim=}!"
'x.ndim=1, zi.ndim=2!'
>>> f"x.ndim={x.ndim}, zi.ndim={zi.ndim}!"
'x.ndim=1, zi.ndim=2!'
```